frontend/src/api/Socket.ts

```diff
@@ -1,54 +1,92 @@
+/* eslint-disable consistent-return */
 import SockJS from 'sockjs-client';
 import Stomp, { Client, Message } from 'stompjs';
 
 let stompClient: Client;
 
-type User = {
+export type User = {
   nickname: string;
 }
 
 // Variable to hold the current connected state.
-let connected = false;
+let connected: boolean = false;
 
 // Create constants for the connection, subscription, and send message URLs.
 export const SOCKET_ENDPOINT:string = '/api/v1/socket/join-room-endpoint';
 const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
-const SEND_USER_URL:string = '/api/v1/socket/user';
+const ADD_USER_URL:string = '/api/v1/socket/add-user';
+const DELETE_USER_URL:string = '/api/v1/socket/delete-user';
 
-export const sendUser = (nickname:string) => {
-  if (connected) {
-    stompClient.send(SEND_USER_URL, {}, nickname || 'Anonymous');
+export const isValidNickname = (nickname: string) => nickname.length > 0
+  && !nickname.includes(' ') && nickname.length <= 16;
+
+/**
+ * Add the user by sending a message via socket.
+ * @returns error, if present
+*/
+export const addUser = (nickname:string): Error | undefined => {
+  if (connected && isValidNickname(nickname)) {
+    stompClient.send(ADD_USER_URL, {}, nickname);
+  } else if (!connected) {
+    return new Error('The socket is not connected.');
   } else {
-    console.error('You must be connected to a socket before sending user information.');
+    return new Error('The provided nickname is invalid.');
   }
 };
```

Comment:
Suggestion: For addUser and deleteUser, I'm not sure if you actually need nickname error validation since you already validate it in `Join.tsx`. Although assuming that in the future we add server-side verification on inputs, then maybe we can keep the error validation here but instead of validating and returning a custom Error, just returning the ErrorResponse received from the server.

Reply:
Yes, my thinking was that somebody could enable the frontend "Enter" button through inspect element even though the current nickname is invalid; in that case, we would still want to catch the case and return an error.

I'm happy to add validation on the server side as well, that makes sense; however, if I keep the validation here as well, what should I do if the nickname is not valid? Should I return some sort of error in that case, or should I just not have this validation at all?

Reply:
After our conversation, I removed the secondary check on the frontend -- done!